app/src/main/java/org/fossasia/openevent/general/SearchResultsActivity.kt

```diff
@@ -0,0 +1,119 @@
+package org.fossasia.openevent.general
+
+import android.arch.lifecycle.Observer
+import android.content.Intent
+import android.support.v7.app.AppCompatActivity
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.text.TextUtils
+import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_search_results.*
+import org.fossasia.openevent.general.event.*
+import org.fossasia.openevent.general.favorite.FavoriteEventsRecyclerAdapter
+import org.fossasia.openevent.general.search.DATE
+import org.fossasia.openevent.general.search.LOCATION
+import org.fossasia.openevent.general.search.QUERY
+import org.fossasia.openevent.general.search.SearchViewModel
+import org.fossasia.openevent.general.utils.Utils
+import org.fossasia.openevent.general.utils.nullToEmpty
+import org.koin.android.architecture.ext.viewModel
+import timber.log.Timber
+
+class SearchResultsActivity : AppCompatActivity() {
+    private val eventsRecyclerAdapter: FavoriteEventsRecyclerAdapter = FavoriteEventsRecyclerAdapter()
+    private val searchViewModel by viewModel<SearchViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_search_results)
+
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.title = resources.getString(R.string.search_results)
+
+        progressBar.isIndeterminate = true
+        eventsRecycler.layoutManager = LinearLayoutManager(this)
+
+        eventsRecycler.adapter = eventsRecyclerAdapter
+        eventsRecycler.isNestedScrollingEnabled = false
+
+        val intent = intent
```

Comment:
Why?

Reply:
> Why?

Okay, I can be directly use `performSearch(intent)`.